src/main/java/org/dependencytrack/event/kafka/processor/VulnerabilityScanResultProcessor.java

```diff
@@ -904,6 +926,18 @@ ON CONFLICT ("VULNERABILITY_ID", "COMPONENT_ID", "PROJECT_ID") DO UPDATE
                 """)
         void createAnalysisComments(@BindMethods final Iterable<AnalysisComment> comment);
 
+        @SqlQuery("""
+                UPDATE "ANALYSIS"
+                    SET "SUPPRESSED" = true, "STATE" = 'NOT_AFFECTED'
+                    WHERE "COMPONENT_ID" = :componentId
+                    AND "VULNERABILITY_ID" IN
+                        ( SELECT "VULNERABILITY"."ID"
+                            FROM "VULNERABILITY"
+                            WHERE "VULNERABILITY"."SOURCE" = 'SNYK')
+                    RETURNING "ANALYSIS"."ID"
+            """)
+        @RegisterBeanMapper(org.dependencytrack.model.Analysis.class)
+        List<Long> suppressSnykAnalyses(@Bind final long componentId);
```

Comment:
This suppresses all Snyk vulnerabilities of a component, but what if only one out of `N` vulnerabilities were rejected? Lots of components have multiple vulnerabilities.

Reply:
Also consider that no analysis may exist already, so an insert would be necessary rather than an update.

Reply:
If Snyk is reporting no results for a component purl then it should be that Snyk doesn't have any vulnerabilities for the component.
Yes, we can insert new analysis instead of update 👍🏼

Reply:
> If Snyk is reporting no results for a component purl then it should be that Snyk doesn't have any vulnerabilities for the component.

Yeah that's fair. But what if Snyk still reports vulnerabilities, but only a *subset* of what it had previously reported?

Reply:
At apiserver end, we can compare the incoming list of vulns for component with it's existing Snyk vulns and suppress the ones which are no longer being reported?